### PR TITLE
Update to testing process

### PR DESCRIPTION
**Pull request by nfearnley**
Moved coveralls and flake8 install to dev-requirements.txt
Moved coverage parameters to .coveragerc config file
Added .pytest_cache to .gitignore
Updated "willie" to "sopel" in comments
Added dev extras_require to setup.py

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,6 +5,7 @@ MANIFEST
 doc/build/*
 logs/*
 tests
+.pytest_cache
```
Comment (dgw) [inline .gitignore]: This is already included below, next to `.cache` and `.coverage`.

File: setup.py [hunk 2]
```diff
@@ -54,5 +54,6 @@ def read_reqs(path):
     license='Eiffel Forum License, version 2',
     platforms='Linux x86, x86-64',
     install_requires=requires,
+    extras_require={'dev': dev_requires},
```
Comment (dgw) [inline setup.py]: This allows doing `pip install sopel[dev]`, right? Nice.

**Comment by nfearnley** [comment]
Yeah. I was trying to look into the best way to separate dev and production
requirements. In the future we might want to do more fine grain
separations, such as docs, test, etc. Or not, depending on needs.

**Comment by nfearnley** [comment]
I noticed that it wasn't ignoring my .pytest-cache, but I was likely
working on a commit before it was added.

**Comment by nfearnley** [comment]
Yes, I see you added it 4 days ago, and the branch I was working in was a
hit older.

**Comment by dgw** [comment]
@nfearnley FYI when you reply by email, Github just dumps your comments into the top-level PR thread instead of whatever line note you responded to. (That's something they should probably work on, tbh. Replying by email is really limited, I've learned.) I know what you're responding to anyway, though. This is good to go once the duplicate gitignore entry is amended out.

**Pull request merged**